lib/src/fnc/util/math/mode.rs

```diff
@@ -1,11 +1,26 @@
 use crate::sql::number::Number;
+use std::collections::HashMap;
 
 pub trait Mode {
+	/// Most Frequent Number in dataset
+	/// O(n*w) time complex s.t. w = distinct count
 	fn mode(self) -> Number;
 }
 
 impl Mode for Vec<Number> {
 	fn mode(self) -> Number {
-		todo!()
+		let frequencies = self.iter().fold(HashMap::new(), |mut freqs, value| {
+			let entry = freqs.entry(value).or_insert(Number::from(0));
+
+			*freqs.entry(value).or_insert(Number::from(0)) = Number::from(1) + entry.clone();
+			freqs
+		});
```

Comment:
Given that floats are hard to hash (as you've noted below), should we use a BTreeMap to store values?

That only requires Ord which is already implemented.

Reply:
TIL BTreeMaps exist in the standard library, implemented! great shout.